### ISSUE TITLE
 Web - Wallet - Error when running code in /settings/wallet/card/expensify page

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#30390]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.91-1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause = Internal Team

**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com/settings/wallet/card/expensify
2. Paste the following code in JS console

Onyx.merge(`cardList`, {
5678: {
cardID: 5678,
state: 4,
bank: '1000',
availableSpend: 10000,
domainName: 'Expensify',
lastFourPAN: '2345',
isVirtual: true,
},
});

## Expected Result:
The code is run successfully

## Actual Result:
An error shows up
VM2042:1 Uncaught ReferenceError: Onyx is not defined
at <anonymous>:1:1

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

 

https://github.com/Expensify/App/assets/78819774/053334c0-c8ef-4daa-8610-5bd697215a3a

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by alitoshmatov** [comment]
I am confused, why this is an issue? Would be great if someone could explain it in more details

**Comment by iiredalert** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Unable to use the Onyx global variable in the console in the staging envrionemnt.

### What is the root cause of that problem?

We are not setting the global Onyx variable for the staging environment, it is only set for the development environment:

https://github.com/Expensify/App/blob/d4d9f78be5d54e18d71b9a6c1881734dea3e7780/src/App.js#L32-L34

### What changes do you think we should make in order to solve the problem?

We can add a check to see if we're in thetsating environment:

```
// For easier debugging and development, when we are in web we expose Onyx to the window, so you can more easily set data into Onyx
if (window && (Environment.isDevelopment() || Environment.IsStaging())) {
    window.Onyx = Onyx;
    window.setSupportToken = Session.setSupportAuthToken;
}
```

Then add the following function in Environment.js:
```
function isStaging() {
    return lodashGet(Config, 'ENVIRONMENT', CONST.ENVIRONMENT.DEV) === CONST.ENVIRONMENT.STAGING;
}
```

### What alternative solutions did you explore? (Optional)

**Comment by stephanieelliott** [comment]
Yeah considering this is not an issue that an end user would run into and we deployed as scheduled, I think this can be closed.
